braindecode/preprocessing/preprocess.py

```diff
@@ -117,11 +123,19 @@ def preprocess(concat_ds, preprocessors):
         A concat of BaseDataset or WindowsDataset datasets to be preprocessed.
     preprocessors: list(Preprocessor)
         List of Preprocessor objects to apply to the dataset.
+    save_dir : str | None
+        If a string, the preprocessed data will be saved under the specified
+        directory.
+    overwrite : bool
+        When `save_dir` is provided, controls whether to overwrite existing
+        files with the same name.
+    n_jobs : int | None
+        Number of jobs for parallel execution.
 
     Returns
     -------
-    BaseConcatDataset:
-        Preprocessed dataset.
+    None | BaseConcatDataset:
```

Comment:
It still feels strange to me to have two different return types. What happens if we return the in-place modified object? Will it be copied?
I would like to always call 
`concat_ds = preprocess(concat_ds, ...)` no matter which arguments are provided. Is this possible? If not, I see the issue that a user might accidentally overwrite his dataset with None and has to repeat all the preprocessing...

Reply:
Actually right now the function doesn't return anything, the docstring was from a previous implementation. 

Good point about overwriting the dataset... I vote for always returning as well.